python/tskit/util.py

```diff
@@ -690,3 +691,32 @@ def random_nucleotides(length: numbers.Number, *, seed: Union[int, None] = None)
     encoded_nucleotides = np.array(list(map(ord, "ACTG")), dtype=np.int8)
     a = rng.choice(encoded_nucleotides, size=int(length))
     return a.tobytes().decode("ascii")
+
+
+def raise_known_file_format_errors(open_file, existing_exception):
+    """
+    Sniffs the file for pk-zip or hdf header bytes, then raises an exception
+    if these are detected, if not raises the existing exception.
+    """
+    # Check for HDF5 header bytes
+    try:
+        open_file.seek(0)
+    except io.UnsupportedOperation:
+        # If we can't seek, we can't sniff the file.
+        raise existing_exception
+    if open_file.read(4) == b"\x89HDF":
+        raise tskit.FileFormatError(
+            "The specified file appears to be in HDF5 format. This file "
+            "may have been generated by msprime < 0.6.0 (June 2018) which "
+            "can no longer be read directly. Please convert to the new "
+            "kastore format using the ``tskit upgrade`` command."
+        )
+    open_file.seek(0)
+    if open_file.read(2) == b"\x50\x4b":
+        raise tskit.FileFormatError(
+            "The specified file appears to be in zip format. Use the tszip "
```

Comment:
I guess someone could just be passing in a standard zipfile, so should be a little careful there?

Reply:
I've couched the statement, now reads `The specified file appears to be in zip format, so may be a compressed tree sequence.`